### PR TITLE
[CDE-883] - Rename folders and reorder components inside CDE editor

### PR DESCRIPTION
**Pull request by joaogameiro**
[CDE-883] - Reorder components inside CDE editor

### CODE REVIEW

File: cde-core/resource/resources/base/components/0100_parameters/0105_TextParameterRender.xml [hunk 1]
```diff
@@ -5,8 +5,9 @@
         <Name>Simple Parameter</Name>
         <IName>Parameter</IName>
         <Description>Simple parameter</Description>
-		<Category>GENERIC</Category>
-		<CatDescription>Generic</CatDescription>
+		<Category>PARAMETERS</Category>
```
Comment (afrjorge) [inline cde-core/resource/resources/base/components/0100_parameters/0105_TextParameterRender.xml]: Can you please fix the incorrect indentations? Same thing for the other files.

Comment (joaogameiro) [inline cde-core/resource/resources/base/components/0100_parameters/0105_TextParameterRender.xml]: Fixed indentation

**Comment by pamval** [comment]
@afrjorge please review this again

**Comment by afrjorge** [comment]
@pamval @joaogameiro I think we shouldn't be moving the component files to locations related to part of the XML configuration. This PR should be similar to https://github.com/webdetails/sparkl/pull/63 and only update the XML content so the editor shows the components in the new places.

**Comment by joaogameiro** [comment]
Number of folder and files is to guarantee order inside CDE editor, because order property is not working.

**Comment by afrjorge** [comment]
I need some time to double check if and why are the XML configuration files' `order` element not being respected. This PR cannot be merged now because moving the custom component's Javascript resources will break the RequireJS configuration.

**Comment by afrjorge** [comment]
Covered in https://github.com/webdetails/cde/pull/766

**Pull request closed without merging**